### PR TITLE
Add walrus operator

### PR DESCRIPTION
**Pull request by mo1ein**
## Change Summary
Use walrus operator that removes extra lines.

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by samuelcolvin** [comment]
We still support 3.7, so can't use walrus operators.

**Comment by mo1ein** [comment]
Ok. but I fixed the typo too. :(

**Comment by hramezani** [comment]
> Ok. but I fixed the typo too. :(

Please update your PR to contain only the typo fix.

File: pydantic/json_schema.py [hunk 1]
```diff
@@ -2017,8 +2017,7 @@ def handle_invalid_for_json_schema(self, schema: CoreSchemaOrField, error_info:
 
     def emit_warning(self, kind: JsonSchemaWarningKind, detail: str) -> None:
         """This method simply emits PydanticJsonSchemaWarnings based on handling in the `warning_message` method."""
-        message = self.render_warning_message(kind, detail)
-        if message is not None:
+        if (message := self.render_warning_message(kind, detail)) is not None:
```
Comment (hramezani) [inline pydantic/json_schema.py:2020]:
```suggestion
        message = self.render_warning_message(kind, detail)
        if message is not None:
```

**Comment by progsmile** [comment]
Imho, walrus operator decreases code readability by making separate statements one-liner. It makes you stuck more on `if` statements (I mean not in this case, but in general)

**Comment by hramezani** [comment]
@mo1ein Do you want to update your PR and keep the typo fix? Otherwise, I will close the PR

**Comment by samuelcolvin** [comment]
Time waster or bot.

**Pull request closed without merging**